web/src/components/Button.tsx

```diff
@@ -22,10 +25,11 @@ export function Button({
   foregroundClasses, // textColor, etc.
   size = 'lg',
   type = 'primary',
+  asDiv, // If true, renders a div instead of a button to avoid nested buttons in components like ToastPrimitive.Action
```

Comment:
Not super happy this was needed but it seemed to be the only way to avoid nested buttons while still re-using our buttons. Happy to hear about other ideas to solve it if you guys have them though.

Reply:
An alternative would be to use the Radix Button component, which has a nice pattern for solving it:

![Screenshot 2024-06-20 at 11 15 39](https://github.com/electricitymaps/electricitymaps-contrib/assets/3296643/c5723e2b-9b16-4122-9fd9-f47630e58c21)

https://www.radix-ui.com/themes/docs/components/button

But that does require some more work though, dunno if we want to do that now :)

Reply:
I tried that first and it messed up all our styling and there was seemingly no easy way to fix it unfortunately. Looked even worse than before at one point...

Reply:
Oh okay, thanks for explaining. Let's roll with your solution then. In the future we could make it a bit more generic with a `as` prop that allows rendering it as whatever, but for now I think this is okay :)